agent/tests/test_lib_aws_sdk_php.c

```diff
@@ -0,0 +1,176 @@
+
+  nr_free(source);
+}
+
+static void test_aws_sdk_php_add_supportability_metric(void) {
+
+  /*
+   * Should return aws metric with classname
+   */
+  tlib_php_request_start();
+
+  int num_metrics = nrm_table_size(NRPRG(txn)->unscoped_metrics);
+  lib_aws_sdk_php_add_supportability_metric(NULL);
+  tlib_pass_if_int_equal("aws supportability metric 0: metric not created in NULL metrics", num_metrics,
+                         nrm_table_size(NRPRG(txn)->unscoped_metrics));
+
+  lib_aws_sdk_php_add_supportability_metric("one\\two");
+  tlib_pass_if_not_null("aws supportability metric 1: service/client metric created",
+                        nrm_find(NRPRG(txn)->unscoped_metrics, PHP_AWS_CLASS_PREFIX
+                                 "one_two"));
+
+  lib_aws_sdk_php_add_supportability_metric("three\\\\four");
+  tlib_pass_if_not_null("aws supportability metric 2: service/client metric created",
+                        nrm_find(NRPRG(txn)->unscoped_metrics, PHP_AWS_CLASS_PREFIX
+                                 "three_four"));  
+
+  lib_aws_sdk_php_add_supportability_metric("three\\four\\five");
+  tlib_pass_if_not_null("aws supportability metric 3: service/client metric created",
+                        nrm_find(NRPRG(txn)->unscoped_metrics, PHP_AWS_CLASS_PREFIX
+                                 "three_four_five"));   
+
+  lib_aws_sdk_php_add_supportability_metric("three\\");
+  tlib_pass_if_not_null("aws supportability metric 4: service/client metric created",
+                        nrm_find(NRPRG(txn)->unscoped_metrics, PHP_AWS_CLASS_PREFIX
+                                 "three"));    
+  lib_aws_sdk_php_add_supportability_metric("\\four");
+  tlib_pass_if_not_null("aws supportability metric 5: service/client metric created",
+                        nrm_find(NRPRG(txn)->unscoped_metrics, PHP_AWS_CLASS_PREFIX
+                                 "four"));                                 
+  lib_aws_sdk_php_add_supportability_metric("five");
+  tlib_pass_if_not_null("aws supportability metric 6: service/client metric created",
+                        nrm_find(NRPRG(txn)->unscoped_metrics, PHP_AWS_CLASS_PREFIX
+                                 "five"));   
+                                                                                                                       
+  tlib_php_request_end();
+}
+
+static void test_aws_sdk_php_handle_version(void) {
+  /*
+   * If lib_aws_sdk_php_handle_version function is ever called, we have already
+   * detected the aws-sdk-php library.
+   */
+
+
+  /*
+   * Aws/Sdk exists. Should return aws package metric with
+   * version
+   */
+  tlib_php_request_start();
+  declare_aws_sdk_class("Aws", "Sdk", LIBRARY_MAJOR_VERSION);
+  lib_aws_sdk_php_handle_version();
+  tlib_pass_if_not_null("version test 1: package metric created",
+                        nrm_find(NRPRG(txn)->unscoped_metrics, PACKAGE_METRIC
+                                 "/" LIBRARY_MAJOR_VERSION "/detected"));
```

Comment:
It is a matter of style - but if the test case input is represented by a `#define` it seems like it would be a good idea for the expected result would also have a similarly named `#define` so if the case is modified it is clear how to handle input and expected result.

Reply:
Can you explain a little more? I'm not quite catching the meaning since the define is already used in the input and output, similar to what was done for the [fw_support tests](https://github.com/newrelic/newrelic-php-agent/blob/main/agent/tests/test_fw_support.c#L74).
The input has a define [here](https://github.com/newrelic/newrelic-php-agent/pull/932/files/0e4f2819dc2fb21bca1867804c09c6a213f10fc0#diff-5cbabeb350f68acfaa71e353ed2387f1d2aa71b7b03201d820a283c3810e3aa2R98), and the output has the define [here](https://github.com/newrelic/newrelic-php-agent/pull/932/files/0e4f2819dc2fb21bca1867804c09c6a213f10fc0#diff-5cbabeb350f68acfaa71e353ed2387f1d2aa71b7b03201d820a283c3810e3aa2R102)

Reply:
Please see tests in  [1148df3](https://github.com/newrelic/newrelic-php-agent/pull/932/commits/1148df3439920382b561b616ba412a7b3a023c08) 

Is that more along the lines of what you meant?